### PR TITLE
List of content. First draft

### PR DESCRIPTION
**Pull request by ktonon**
Screen Shot 2020-02-12 at 9 46 37 AM

Unfinished:

- responsive layout
- sortable headers
- paging
- button handlers
- mapping for all content types
- relative date formatting (ex, 2 hours ago)

### CODE REVIEW

File: src/components/relative-date.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+	static get styles() {
+		return [css`
+		`];
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		const date = new Date(Date.parse(this.value));
+		this.formatted = formatDate(date, { format: 'short' });
```
Comment (ktonon) [inline src/components/relative-date.js]: Plan to extend this to handle relative dates as shown in the mockup.

File: src/util/content-type.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			return ['carnelian', 'white'];
+	}
+};
+
+export const typeLocalizationKey = type => {
+	switch (type.toLowerCase()) {
+		case 'scorm':
+			return fullKey('Scorm');
+		case 'video':
+			return fullKey('VideoNote');
```
Comment (ktonon) [inline src/util/content-type.js]: Will extend this to uses client app to determine purposes of document.

File: src/my-objects.js [hunk 3]
```diff
@@ -1,28 +1,142 @@
...
 	render() {
 		return html`
-			<h2>My Objects</h2>
-			<p>API endpoint: ${this.apiClient.endpoint}</p>
-			<p>Tenant: ${this.apiClient.tenantId}</p>
-			<file-uploader></file-uploader>
+			<h2 class="d2l-heading-2 h2-custom">${this.localize('myContent')}</h2>
+			<d2l-list class="table-header" separators="none">
+				${this.renderTableHeader()}
+			</d2l-list>
```
Comment (ktonon) [inline src/my-objects.js]: Using another `<d2l-list>` for the column headers to get them to align. Had to fudge the checkbox width (see above; `42px` wide)

File: src/my-objects.js [hunk 4]
```diff
@@ -1,28 +1,142 @@
...
+
+	constructor() {
+		super();
+		this.contentItems = [];
+		this.contentItemsSelectable = true;
+	}
+
 	connectedCallback() {
 		super.connectedCallback();
 		this.apiClient = this.requestDependency('content-service-client');
```
Comment (qwangd2l) [inline src/my-objects.js]: For the purpose of unit axe test, do you think we should do a `if (this.apiClient) { this.loadPage() }` and supply mock `contentItems` and pass them as an attribute in the unit test as a fixture ?

**Comment by rcausey** [comment]
Looks great. Can you spend the rest of the sprint working on the short dates and implement a dummy dropdown that opens when clicking the '...'?

**Pull request merged**